components/support/nimbus-fml/src/workflows.rs

```diff
@@ -149,4 +177,44 @@ mod test {
         generate_and_assert("test/app_menu.kts", "fixtures/ir/app_menu.json", true)?;
         Ok(())
     }
+
+    fn validate_against_experimenter_schema<P: AsRef<Path>>(
+        schema_path: P,
+        generated_json: &serde_json::Value,
+    ) -> Result<()> {
+        let schema = fs::read_to_string(&schema_path)?;
+        let schema: serde_json::Value = serde_json::from_str(&schema)?;
+        let compiled = JSONSchema::compile(&schema).expect("The schema is invalid");
+        let res = compiled.validate(generated_json);
+        if let Err(e) = res {
+            let mut errs: String = "Validation errors: \n".into();
+            for err in e {
+                errs.push_str(&format!("{}\n", err));
+            }
+            panic!("{}", errs);
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_schema_validation() -> Result<()> {
+        for path in MANIFEST_PATHS {
+            let out_tmpdir = TempDir::new("schema_validation").unwrap();
+            let manifest_fml = join(pkg_dir(), path);
+            let curr_out = out_tmpdir.as_ref().join(path.split('/').last().unwrap());
+            let cmd = GenerateExperimenterManifestCmd {
+                manifest: manifest_fml.into(),
+                output: curr_out.clone(),
+                load_from_ir: true,
+            };
+            generate_experimenter_manifest(None, cmd)?;
+            let generated = fs::read_to_string(curr_out)?;
+            let generated_json = serde_json::from_str(&generated)?;
+            validate_against_experimenter_schema(
+                join(pkg_dir(), "ExperimentFeatureManifest.schema.json"),
+                &generated_json,
+            )?;
```

Comment:
Oh I see, this looks like a reasonably straightforward optimization. (Good second bug perhaps)